### PR TITLE
Pbi 455 eui organ carousel

### PR DESCRIPTION
**Pull request by edlu77**
#455

### CODE REVIEW

File: projects/ccf-shared/src/lib/components/organ-selector/organ-selector.component.ts [hunk 1]
```diff
@@ -121,6 +121,7 @@ export class OrganSelectorComponent implements AfterViewInit, OnDestroy {
    * Set resize sensor on carousel
    */
   ngAfterViewInit(): void {
+    this.selectedOrgans = this.selectedOrgans.slice(1);
```
Comment (bherr2) [inline projects/ccf-shared/src/lib/components/organ-selector/organ-selector.component.ts:124]: What is this?

Comment (edlu77) [reply projects/ccf-shared/src/lib/components/organ-selector/organ-selector.component.ts:124]: It's to deselect the first item by default. Should I do it some other way?

Comment (bherr2) [reply projects/ccf-shared/src/lib/components/organ-selector/organ-selector.component.ts:124]: Why would you do that?

File: projects/ccf-eui/src/app/app.component.html [hunk 2]
```diff
@@ -27,7 +27,16 @@
       <ccf-drawer-toggle-button></ccf-drawer-toggle-button>
     </ccf-drawer>
     <ccf-drawer-content>
-      <!-- <ccf-organ-selector [multiselect]="true" [organList]="organList" [selectedOrgans]="[organList[1], organList[2], organList[3]]"></ccf-organ-selector> -->
+      <div [class.closed]="!open" class="selector-drawer">
+        <ccf-organ-selector [multiselect]="true" [organList]="organList" [selectedOrgans]="selectedOrgans"></ccf-organ-selector>
```
Comment (bherr2) [inline projects/ccf-eui/src/app/app.component.html:31]: When selected organs change in ccf-organ-selector, save the new selected organs back out to `selectedOrgans` field in app.componenent.ts

**Review by bherr2 — CHANGES_REQUESTED** [review]

**Review by bherr2 — CHANGES_REQUESTED** [review]
Two things: the styling of selected organs don't fit the current spec and I don't see counts on the organs in the carousel. Fine to set those to zero for the purposes of this PR.

File: projects/ccf-eui/src/app/app.component.ts [hunk 3]
```diff
@@ -25,12 +25,22 @@ export class AppComponent {
...
   /**
    * Used to keep track of the ontology label to be passed down to the
    * results-browser component.
    */
   ontologySelectionLabel = 'body';
 
+  /**
+   * Whether or not organ carousel is open
+   */
+  open = true;
```
Comment (bherr2) [inline projects/ccf-eui/src/app/app.component.ts:42]: let's refactor this to say `organListVisible`. It'll eventually fall into the state, but will make it more clear.

File: projects/ccf-shared/src/lib/components/organ-selector/organ-selector.component.html [hunk 4]
```diff
@@ -1,11 +1,14 @@
...
   <div class="carousel-item-container" #itemcontainer>
-    <div class="carousel-item-list" [class.hidden]="hidden" #itemlist>
+    <div class="carousel-item-list" #itemlist>
       <div *ngFor="let organ of organList" class="carousel-item" [class.disabled]="organ.disabled" (click)="selectOrgan(organ)"
           [class.selected]="isSelected(organ)">
           <mat-icon class="carousel-icon" svgIcon="{{organ.src}}"></mat-icon>
-          <span class="carousel-text">{{organ.name}}</span>
+          <div class="carousel-text-container">
+            <div class="carousel-text">{{ organ.name }}</div>
+            <div class="organ-number" *ngIf="multiselect">{{ organ.num ? organ.num : 0 }}</div>
```
Comment (bherr2) [inline projects/ccf-shared/src/lib/components/organ-selector/organ-selector.component.html:10]: `*ngIf="organ.num === undefined"`

File: projects/ccf-shared/src/lib/components/organ-selector/organ-selector.component.scss [hunk 5]
```diff
@@ -52,13 +47,30 @@
...
+              display: flex;
+              justify-content: center;
+              align-items: center;
+              font-size: 0.625rem;
+              max-width: 3.5rem;
+              line-height: .8rem;
+              min-height: 1.5rem;
+            }
+  
+            .organ-number {
```
Comment (bherr2) [inline projects/ccf-shared/src/lib/components/organ-selector/organ-selector.component.scss:67]: Is there a better class name/way to describe extra text below the carousel item?

File: projects/ccf-shared/src/lib/components/organ-selector/organ-selector.component.ts [hunk 6]
```diff
@@ -51,6 +51,8 @@ export interface OrganInfo {
    * True if applies to both sexes
    */
   hasSex?: boolean;
+
+  num?: number;
```
Comment (bherr2) [inline projects/ccf-shared/src/lib/components/organ-selector/organ-selector.component.ts:55]: num => numResults

**Review by bherr2 — CHANGES_REQUESTED** [review]

File: projects/ccf-shared/src/lib/components/organ-selector/organ-selector.component.scss (hunk 5 above)
Comment (edlu77) [reply projects/ccf-shared/src/lib/components/organ-selector/organ-selector.component.scss:67]: Would donor-registrations work?

Comment (bherr2) [reply projects/ccf-shared/src/lib/components/organ-selector/organ-selector.component.scss:67]: I was thinking something more generic like num-results

File: projects/ccf-shared/src/lib/components/organ-selector/organ-selector.component.html [hunk 7]
```diff
@@ -1,11 +1,14 @@
...
   <div class="carousel-item-container" #itemcontainer>
-    <div class="carousel-item-list" [class.hidden]="hidden" #itemlist>
+    <div class="carousel-item-list" #itemlist>
       <div *ngFor="let organ of organList" class="carousel-item" [class.disabled]="organ.disabled" (click)="selectOrgan(organ)"
           [class.selected]="isSelected(organ)">
           <mat-icon class="carousel-icon" svgIcon="{{organ.src}}"></mat-icon>
-          <span class="carousel-text">{{organ.name}}</span>
+          <div class="carousel-text-container">
+            <div class="carousel-text">{{ organ.name }}</div>
+            <div class="num-results" *ngIf="organ.numResults !== undefined">{{ organ.numResults ? organ.numResults : 0 }}</div>
```
Comment (bherr2) [inline projects/ccf-shared/src/lib/components/organ-selector/organ-selector.component.html:10]: You shouldn't need the extract tertiary logic now that we're hiding when numResults is undefined.

File: projects/ccf-shared/src/lib/components/organ-selector/organ-selector.component.ts [hunk 8]
```diff
@@ -2,22 +2,22 @@ import { AfterViewInit, OnDestroy, Component, EventEmitter, HostBinding, Input,
...
-  { src: 'app:kidney-left', organ: 'Kidney', name: 'Kidney, L', side: 'left', hasSex: true },
-  { src: 'app:kidney-right', organ: 'Kidney', name: 'Kidney, R', side: 'right', hasSex: true },
-  { src: 'app:spleen', organ: 'Spleen', name: 'Spleen', hasSex: true },
-  { src: 'app:bladder', organ: 'Bladder', name: 'Bladder', disabled: true, hasSex: true },
-  { src: 'app:brain', organ: 'Brain', name: 'Brain', disabled: true, hasSex: true },
-  { src: 'app:liver', organ: 'Liver', name: 'Liver', disabled: true, hasSex: true },
-  { src: 'app:lung-left', organ: 'Lung', name: 'Lung, L', disabled: true, side: 'left', hasSex: true },
-  { src: 'app:lung-right', organ: 'Lung', name: 'Lung, R', disabled: true, side: 'right', hasSex: true },
-  { src: 'app:lymph_nodes', organ: 'Lymph Nodes', name: 'Lymph Nodes', disabled: true, hasSex: true },
+  { src: 'app:large_intestine', organ: 'Large Intestine', name: 'Large Intestine', hasSex: true, numResults: 0 },
```
Comment (bherr2) [inline projects/ccf-shared/src/lib/components/organ-selector/organ-selector.component.ts:5]: Did you test this in the RUI? This seems like it'll show results in there as well... In the business logic in the EUI put this where you are also filtering the data initially: (ie `filter().map(n => { ...n, n.numResults: 0 }`)

**Review by bherr2 — APPROVED** [review]

**Pull request merged**